lib/buf.gen.yaml

```diff
@@ -0,0 +1,8 @@
+version: v2
+
+clean: true
+plugins:
+  - local: protoc-gen-es
+    out: src/platform
+    include_imports: true
+    opt: target=ts,import_extension=.js
```

Comment:
TODO: Add a comment here for `target=ts`

The consumers do need  `js+dts` so that the library can be used just with javascript. They are getting that, here is why:

Internally `ts` files are generated. When a build is made, those generated `ts` files are compiled in `dist/lib/src/platform` into  `js` + `dts`.
So for the library itself, we use generate and use the `ts` definitions. For consumers after the build, its `js + dts`